code/modules/assembly/flash.dm

```diff
@@ -205,6 +205,9 @@
 	if(victim.loc == attacker.loc)
 		return DEVIATION_PARTIAL
 
+	if(HAS_TRAIT(victim, TRAIT_LIGHT_SENSITIVE))
+		return DEVIATION_NONE
+
```

Comment:
If they have 360 degree vision, shouldn't this check be above the same turf check?